test/stdlib/rubygems/Gem_test.rb

```diff
@@ -0,0 +1,597 @@
+    assert_send_type  "() -> String",
+                      Gem, :find_config_file
+  end
+
+  def test_find_files
+    assert_send_type  "(String) -> Array[String]",
+                      Gem, :find_files, "fileutils.rb"
+    assert_send_type  "(String, bool) -> Array[String]",
+                      Gem, :find_files, "fileutils.rb", true
+  end
+
+  def test_find_latest_files
+    assert_send_type  "(String) -> Array[String]",
+                      Gem, :find_latest_files, "fileutils.rb"
+    assert_send_type  "(String, bool) -> Array[String]",
+                      Gem, :find_latest_files, "fileutils.rb", false
+  end
+
+  def test_find_unresolved_default_spec
+    assert_send_type  "(String) -> Gem::Specification?",
+                      Gem, :find_unresolved_default_spec, "fileutils.rb"
+  end
+
+  def test_finish_resolve
+    assert_send_type  "() -> Array[Gem::Specification]",
+                      Gem, :finish_resolve
+    assert_send_type  "(Gem::RequestSet) -> Array[Gem::Specification]",
+                      Gem, :finish_resolve, Gem::RequestSet.new(Gem::Dependency.new("pg"))
+  end
+
+  def test_gemdeps
+    assert_send_type  "() -> Gem::RequestSet::GemDependencyAPI?",
+                      Gem, :gemdeps
+  end
+
+  def test_host
+    assert_send_type  "() -> String",
+                      Gem, :host
+  end
+
+  def test_host=
+    assert_send_type  "(String) -> String",
+                      Gem, :host=, "foo"
+  end
+
+  def test_install
+    omit "due to side-effect"
+    assert_send_type  "(String, Gem::Requirement) -> Array[Gem::Specification]",
+                      Gem, :install, "", Gem::Requirement.default
+  end
+
+  def test_java_platform?
+    assert_send_type  "() -> bool",
+                      Gem, :java_platform?
+  end
+
+  def test_latest_rubygems_version
+    assert_send_type  "() -> Gem::Version",
+                      Gem, :latest_rubygems_version
+  end
+
+  def test_latest_spec_for
+    assert_send_type  "(String) -> nil",
+                      Gem, :latest_spec_for, ""
+    assert_send_type  "(String) -> Gem::Specification",
+                      Gem, :latest_spec_for, "fileutils"
+  end
+
+  def test_latest_version_for
+    assert_send_type  "(String) -> nil",
+                      Gem, :latest_version_for, ""
+    assert_send_type  "(String) -> Gem::Version",
+                      Gem, :latest_version_for, "fileutils"
+  end
+
+  def test_load_env_plugins
+    assert_send_type  "() -> Array[String]",
+                      Gem, :load_env_plugins
+  end
+
+  def test_load_path_insert_index
+    assert_send_type  "() -> Integer",
+                      Gem, :load_path_insert_index
+  end
+
+  def test_load_plugins
+    assert_send_type  "() -> Array[String]",
+                      Gem, :load_plugins
+  end
+
+  def test_load_yaml
+    assert_send_type  "() -> bool?",
+                      Gem, :load_yaml
+  end
+
+  def test_loaded_specs
+    omit "due to Bundler::StubSpecification returns"
```

Comment:
[note] `Gem.loaded_specs` returns a `Hash` including `Bundler::StubSpecification`, but this test fails because `Bundler::StubSpecification` is **not** a `Gem::BasicSpecification`.

```console
$ bundle exec ruby -e 'pp Bundler::StubSpecification.ancestors'
[Bundler::StubSpecification,
 Bundler::RemoteSpecification,
 Comparable,
 Bundler::MatchPlatform,
 Bundler::GemHelpers,
 Object,
 Kernel,
 BasicObject]
```

Please tell me a better way if you know it. 🙏

Reply:
Looks good to skip testing. 🎿